tests/src/kinAccount.unit.test.ts

```diff
@@ -288,25 +342,173 @@ describe("KinAccount.createAccount", async () => {
 		expect(await kinAccount.submitTransaction(txBuilder)).toEqual("708ebb9e3c8890333daca3faa6707b89dc0155f2578314e302f39e7387d2d07c");
 	});
 
+	test("retry send kin, change fee before submitting transaction.", async () => {
+		mockLoadAccountResponse("6319125253062662");
+		mockRetry();
+
+		const txBuilder = await kinAccount.buildSendKin({
+			address: receiverPublic,
+			amount: 23.3,
+			fee: amount,
+			memoText: memo
+		});
+		txBuilder.addFee(20);
+		expect(await kinAccount.submitTransaction(txBuilder)).toEqual("708ebb9e3c8890333daca3faa6707b89dc0155f2578314e302f39e7387d2d07c");
+		expect(retryDelay).toEqual(144);
+	});
+
+	test("retry send kin, error 500 ServerError.", async () => {
+		mockLoadAccountResponse("6319125253062662");
+		mockFailedRetry();
+
+		const txBuilder = await kinAccount.buildSendKin({
+			address: receiverPublic,
+			amount: 23.3,
+			fee: amount,
+			memoText: memo
+		});
+		txBuilder.addFee(20);
+		await expect(kinAccount.submitTransaction(txBuilder)).rejects.toEqual(ErrorDecoder.translate({ response: mock500NetworkResponse }));
+	});
 });
 
-function mock400SendKinResponse(response: ErrorResponse) {
-	nock(fakeUrl)
+function mockRetry() {
+	nock(horizonUrl)
 		.matchHeader(headerKey, headerVal)
-		.post(url => url.includes("/transactions"), /tx=\w+/gi)
-		.reply(response.status, response);
+		.post(url => true)
+		.reply(mock500NetworkResponse.status, mock500NetworkResponse)
+		.post(url => true)
+		.reply(mock500NetworkResponse.status, mock500NetworkResponse)
+		.post(url => true)
+		.reply(mock500NetworkResponse.status, mock500NetworkResponse)
+		.post(url => url.includes("/transactions"), "tx=AAAAAG809%2BMhGZ82rRmsoUDGVlkQGcjGXbF2fX62aTPsrih8AAAAFAAWczYAAAAHAAAAAAAAAAEAAAAQMS1hYWF" +
```

Comment:
this tx envelope is shared among multiple methods, make it const for better readability.